file-formats/uipg/type/zif_aff_uipg_v1.intf.abap

```diff
@@ -0,0 +1,169 @@
+INTERFACE zif_aff_uipg_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">General Information</p>
+    "! General information
+    BEGIN OF ty_general_information,
+      "! <p class="shorttext">Title</p>
+      "! Title of the page template
+      "! $required
+      title       TYPE zif_aff_types_v1=>ty_description_100,
+      "! <p class="shorttext">Description</p>
+      "! Additional description
+      description TYPE zif_aff_types_v1=>ty_description_100,
+      "! <p class="shorttext">Merge ID</p>
+      "! Merge ID
+      merge_id    TYPE c LENGTH 35,
+    END OF ty_general_information.
+
+  "! <p class="shorttext">Catalog Type</p>
+  "! Catalog type
+  "! $values {@link zif_aff_uipg_v1.data:co_catalog_type}
+  "! $default {@link zif_aff_uipg_v1.data:co_catalog_type.standard}
+  TYPES ty_catalog_type TYPE c LENGTH 3.
+
+  CONSTANTS:
+    "! <p class="shorttext">Catalog Type</p>
+    "! catalog type
+    BEGIN OF co_catalog_type,
+      "! <p class="shorttext">Standard Catalog</p>
+      "! Standard catalog
+      "! $enumValue 'CAT'
```

Comment:
I removed the enum value altogether.

Only restricting it to CAT could have had potential issues to begin with and if I have a enum with a single value the schema would already require it to allways have this value. I chose to know more directly reflect the DB values, but since the value is typically static I made it a default value, so that the json can ignore it.

- removed the enum
- set the default value to 'CAT'
- mode tile and catalog key subobjects to more accurately capture the relation between the different key properties

The example JSON atleast looks much nicer :)

```      {
          "id": "00O2TIH53H32S6MBXW7CNIII1",
          "tileAssignment": {
            "tile": {
              "catalogId": "/UI2/FLP_DEMO_TILE_TYPES",
              "id": "00O2TIH53H32KCRLQQ0N2LS33"
            },
            "targetMapping": {
              "catalogId": "/UI2/FLP_DEMO_SAMPLEAPPS_UI5DK",
              "id": "00O2TR99M0M2CFROG2SBFAVCC"
            }
          }
        }
```

Reply:
> set the default value to 'CAT'

I hope my suggestion was not misleading. I guess I would have used "standard" as default, if you introduce an enum later. "CAT" sounds to me like a really technical value. Let me know when we should talk about it in more detail...

Reply:
See comment somewhere below